### ISSUE TITLE
Retrieving OAuth 2.0 tokens via Out Of Band redirect_uri is deprecated

### ISSUE THREAD

**Issue by EmperorDali** [dengste/org-caldav#249]
tl;dr: [Google has deprecated](https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#disallowed-oob) the Out Of Band ('`oob`') method of obtaining OAuth 2.0 authentication tokens which is what `org-caldav` uses via the emacs package `oauth2`. This means that it's no longer possible to generate the authentication tokens needed for `org-caldav` to sync with Google Calendar. Existing tokens will work with `oob` flow until the end of October 2022 at which point they will also stop working.

The fix for this is to migrate the OAuth 2.0 flow to `localhost`, which requires running a local webserver that the OAuth2.0 login will redirect to and provide the token to.

[See this related issue in the `org-gcal` repo for more details](https://github.com/kidd/org-gcal.el/issues/191)

This isn't really an `org-caldav` issue but rather something that should be fixed in the emacs `oauth2` package, but nevertheless I wanted to raise the issue here in case folks wonder why they aren't able to configure `org-caldav` to sync against Google Calendar.

**Comment by boogsbunny** [comment]
Is there a plan to fix this?
